pkg/registry/servicecatalog/broker/validation.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
```

Comment:
1. I think I can agree with this statement, provisionally, from an engineering perspective. I would like an example of the end goal someone would want.

2. Isn't the validation specific to storage? Why would I care about the validations for a type separate from anything else? If they're type specific, why doesn't the type have directly on it Validate and ValidateUpdate(obj) methods?

3. For what purpose?

The way most things are structured, I don't see how one could compile the types without pulling in k8s code, if that is to be the goal.